### PR TITLE
US155532 - Introduce LTR/RTL controls for Full Page

### PR DESCRIPTION
**Pull request by PaulKrznaric**
Currently the editor supports LTR/RTL for individual selected attributes via the Attributes button.
This change introduces LTR/RTL controls for the entire page at once. This can be overridden by changing the RTL/LTR of individual sections. 

This is how the new options for the page dialog looks, it is accessible on the pages listed below via More Settings -> Attributes -> Page Attributes
image
image

#### View:
image
#### Edit:
image
#### Source Code:
```<!DOCTYPE html>
<html><head></head><body style="color: rgb(32, 33, 34); font-family: verdana, sans-serif; font-size: 12px;" dir="rtl"><p>I use page formatting</p>
<p></p>
<p dir="ltr">I use selection formatting</p>
<p dir="ltr"></p>
<p>I use default formatting</p>
<p><img src="/content/Monkey%20Typewriter%20Simpsons.jpg" alt="" title="" data-d2l-editor-default-img-style="" style="max-width: 100%;"></p>
<p dir="ltr"></p>
<p dir="ltr"></p></body></html>
```
## Notes:
This has been verified to display correctly in the following places:
* [x] Classic Content
* [x] Lessons
* [x] New Content Experience
* [x] Widgets*

\* It appears that for widgets, separate directions other than the page direction are not being respected, see the screenshot below:**

image

\** Fixed in this PR: https://github.com/Brightspace/brightspace-integration/pull/9873

## Testing Notes:
- In order to implement this change, some changes were made for Selection Attributes Text Direction. Text Direction should be smoke tested. 

## Related PRs:
BSI: https://github.com/Brightspace/brightspace-integration/pull/9873
Launch Darkly: https://github.com/Brightspace/lms-launch-darkly-flags/pull/10265
LMS: https://github.com/Brightspace/lms/pull/39942

### CODE REVIEW

File: cli/tinymce-import-generator.js [hunk 1]
```diff
@@ -20,14 +20,14 @@ function createExport() {
...
-		+ 	`\tcoreEditor: '${baseLibPath}/tinymce/tinymce.min.js',\n`
+		+ 	`\tcoreEditor: '${baseLibPath}/tinymce/tinymce.js',\n`
 		+ 	`\temojis: '${baseLibPath}/emoji/emojis.js',\n`
 		+ 	`\tlangs: '${baseLibPath}/langs/langs.js',\n`
 		+ 	`\tskin: '${baseLibPath}/skins/d2l'\n`
 		+ '};\n'
 		+ 'const previousTinymcePaths = {\n'
 		+ 	`\tbaseImport: '${previousBaseLibPath}',\n`
-		+ 	`\tcoreEditor: '${previousBaseLibPath}/tinymce/tinymce.min.js',\n`
+		+ 	`\tcoreEditor: '${previousBaseLibPath}/tinymce/tinymce.js',\n`
```
Comment (grant-cleary) [inline cli/tinymce-import-generator.js:30]: Undo these changes please.

File: components/plugins/attributes.js [hunk 2]
```diff
@@ -395,6 +420,13 @@ class AttributesDialog extends Localizer(RtlMixin(LitElement)) {
 						.value="${live(this.attributes.pageLanguage || '')}">
 					</d2l-input-text>
 				</div>
+				${this.showPageAttributeTextDirection ? html`
+				<label class="d2l-input-label d2l-skeletize" for="d2l-attributes-direction-select-page">${this.localize('attributes:direction:label')}</label>
+				<select id="d2l-attributes-direction-select-page" class="d2l-input-select" .value="${live(this.attributes.textDirectionPage || directionOptions.Default)}">
+					<option ?selected="${this._getDefaultPageDirection(directionOptions.Default)}" value="${directionOptions.Default}">${this.localize('attributes:direction:default')}</option>
+					<option ?selected="${this._getDefaultPageDirection(directionOptions.LTR)}" value="${directionOptions.LTR}">${this.localize('direction:ltr')}</option>
+					<option ?selected="${this._getDefaultPageDirection(directionOptions.RTL)}" value="${directionOptions.RTL}">${this.localize('direction:rtl')}</option>
+					` : html``}
```
Comment (grant-cleary) [inline components/plugins/attributes.js:429]: Also note that you backup case can just be `''`. No need for a Lit template.

**Review by grant-cleary — APPROVED** [review]

File: components/plugins/attributes.js [hunk 3]
```diff
@@ -78,7 +80,7 @@ export class AttributesPlugin {
 							setAttr('id', e.detail.attributes.id);
 							setAttr('class', e.detail.attributes.className);
 							setAttr('style', e.detail.attributes.style);
-							setAttr('dir', e.detail.attributes.textDirection);
+							setAttr('dir', e.detail.attributes.textDirectionSelect);
```
[the remark this answers is not in the document]
Comment (grant-cleary) [reply components/plugins/attributes.js:83]: `dir` is indeed a standard thing: https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/dir

File: components/plugins/attributes.js [hunk 4]
```diff
@@ -109,6 +111,16 @@ export class AttributesPlugin {
 									editorDocument.documentElement.removeAttribute('lang');
 									hasChanges = true;
 								}
+
+								if (dialog.showPageAttributeTextDirection && e.detail.attributes.textDirectionPage) {
+									if (editorDocument.body.getAttribute('dir') !== e.detail.attributes.textDirectionPage) {
+										editorDocument.body.setAttribute('dir', e.detail.attributes.textDirectionPage);
+										hasChanges = true;
+									}
+								} else if (dialog.showPageAttributeTextDirection && editorDocument.body.hasAttribute('dir')) {
```
Comment (klhuff) [inline components/plugins/attributes.js:120]: Is the implication here that `e.detail.attributes.textDirectionPage` being false means that we should remove the property.

**Review by klhuff — APPROVED** [review]

File: components/plugins/attributes.js (hunk 4 above)
Comment (PaulKrznaric) [reply components/plugins/attributes.js:120]: The implication here is that if it's empty, I.e. the `default` case, we should remove the text direction to reset it back to default.
